homeassistant/components/climate/melissa.py

```diff
@@ -0,0 +1,252 @@
+"""
+Support for Melissa Climate A/C.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.melissa/
+"""
+import logging
+
+from homeassistant.components.climate import ClimateDevice, \
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE, SUPPORT_ON_OFF, \
+    STATE_AUTO, STATE_HEAT, STATE_COOL, STATE_DRY, STATE_FAN_ONLY, \
+    SUPPORT_FAN_MODE
+from homeassistant.components.fan import SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH
+from homeassistant.components.melissa import DATA_MELISSA, DOMAIN
+from homeassistant.const import TEMP_CELSIUS, STATE_ON, STATE_OFF, \
+    STATE_UNKNOWN, STATE_IDLE, ATTR_TEMPERATURE
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE |
+                 SUPPORT_ON_OFF | SUPPORT_FAN_MODE)
+
+ICON = "mdi:thermometer"
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Iterate through and add all Melissa devices."""
+    connection = hass.data[DATA_MELISSA]
+    devices = connection.fetch_devices().values()
+
+    all_devices = []
+
+    for device in devices:
+        all_devices += [
+            MelissaClimate(connection, device['serial_number'], device)]
+
+    if all_devices:
+        add_devices(all_devices)
+
+
+class MelissaClimate(ClimateDevice):
+    """Representation of a Melissa Climate device."""
+
+    def __init__(self, connection, serial_number, init_data):
+        """Initialize the climate device."""
+        self._name = init_data['name']
+        self._connection = connection
+        self._serial_number = serial_number
+        self._data = init_data['controller_log']
+        self._state = None
+        self._cur_settings = None
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return current state."""
+        if not self._cur_settings:
+            return STATE_UNKNOWN
+        return self._cur_settings[self._connection.STATE] in (
+            self._connection.STATE_ON, self._connection.STATE_IDLE)
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return ICON
+
+    @property
+    def current_fan_mode(self):
+        """Return the current fan mode."""
+        if not self._cur_settings:
+            return STATE_UNKNOWN
+        return self.melissa_fan_to_hass(
+            self._cur_settings[self._connection.FAN])
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._data[self._connection.TEMP]
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
```

Comment:
There's `PRECISION_WHOLE` available in `homeassistant.const`, although imho this is saner than having a const for it..